src/main/java/org/embulk/util/aws/credentials/AwsCredentials.java

```diff
@@ -211,16 +211,16 @@ public void refresh() {
                     "'" + accountIdOption + "'");
             final String roleName = require(task.getRoleName(),
                     "'" + roleNameOption + "'");
-            final String externalId = require(task.getExternalId(),
-                    "'" + externalIdOption + "'");
             final String arn = String.format(ARN_PATTERN, task.getArnPartition(), accountId, roleName);
 
             // use AWSSecurityTokenServiceClient with DefaultAWSCredentialsProviderChain
             // https://javadoc.io/doc/com.amazonaws/aws-java-sdk-sts/1.11.0/com/amazonaws/services/securitytoken/AWSSecurityTokenServiceClient.html#AWSSecurityTokenServiceClient()
             STSAssumeRoleSessionCredentialsProvider.Builder builder
                     = new STSAssumeRoleSessionCredentialsProvider.Builder(arn, task.getSessionName());
-            return builder.withExternalId(externalId)
-                    .withRoleSessionDurationSeconds(task.getDurationInSeconds())
+            if (task.getExternalId().isPresent()) {
+                builder.withExternalId(task.getExternalId().get());
+            }
```

Comment:
It basically looks good, so approving, but we may often want to track what were happening there in the real-world operations.

Then how about leaving a log like :

```suggestion
            if (task.getExternalId().isPresent()) {
                log.info("ExternalId is specified with AssumeRole.");
                builder.withExternalId(task.getExternalId().get());
            } else {
                log.info("ExternalId is not specified for AssumeRole.");
            }
```